core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -443,6 +443,40 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  val PARALLEL_PARTITION_DISCOVERY_THRESHOLD =
+    ConfigBuilder("spark.rdd.sources.parallelPartitionDiscovery.threshold")
+      .doc("The maximum number of paths allowed for listing files at driver side. If the number " +
+        "of detected paths exceeds this value during partition discovery, it tries to list the " +
+        "files with another Spark distributed job. This configuration is effective only when " +
+        "using file-based using HadoopFSUtils")
+      .version("1.5.0")
+      .intConf
+      .checkValue(parallel => parallel >= 0, "The maximum number of paths allowed for listing " +
+        "files at driver side must not be negative")
+      .createWithDefault(2)
```

Comment:
The default values of `spark.sql.sources.parallelPartitionDiscovery.threshold` is 32, why is it 32 here?